### PR TITLE
[READY] Roundstart spawning without job landmark puts you on shuttle

### PR DESCRIPTION
**Pull request by JJRcop**
## About The Pull Request

Discovered via paramedics missing their start landmarks, jobs without landmarks would spawn on the last landmark to be loaded in the map. This appears to not have been intended, and they now spawn on the arrivals shuttle if available. If that fails they spawn in the error room.

I am touching job distribution code so I'd like this to be testmerged for a bit before getting merged.

## Why It's Good For The Game

This is a bug in an error condition, so it will only show up if there are map bugs, which is why we haven't uncovered it for so long.

## Changelog
:cl: JJRcop
fix: Jobs missing spawns will spawn on the arrivals shuttle.
/:cl:

### CODE REVIEW

File: code/modules/mob/living/living.dm [hunk 1]
```diff
@@ -1383,3 +1383,11 @@
 			STAMINA:<font size='1'><a href='?_src_=vars;[HrefToken()];mobToDamage=[refid];adjustDamage=stamina' id='stamina'>[getStaminaLoss()]</a>
 		</font>
 	"}
+
+/mob/living/proc/move_to_error_room()
+	var/obj/effect/landmark/error/error_landmark = locate(/obj/effect/landmark/error) in GLOB.landmarks_list
+	if(error_landmark)
+		forceMove(error_landmark.loc)
+	else
+		forceMove(locate(4,4,1)) //Even if the landmark is missing, this should put them in the error room.
```
Comment (optimumtact) [inline code/modules/mob/living/living.dm]: leave a comment saying you're sorry for whoever finds this when it inevitably breaks and also print a stack trace about the landmark missing

Comment (JJRcop) [reply code/modules/mob/living/living.dm]: Isn't a stack trace provided by `CRASH`? The comment is a good idear tho

Comment (cacogen) [reply code/modules/mob/living/living.dm]: if erro's room has an area why not teleport to that

Comment (optimumtact) [reply code/modules/mob/living/living.dm]: I didn't see the crash for some reason

File: code/controllers/subsystem/job.dm [hunk 2]
```diff
@@ -421,22 +421,23 @@ SUBSYSTEM_DEF(job)
...
-				continue
-			if(locate(/mob/living) in sloc.loc)
-				continue
-			S = sloc
-			sloc.used = TRUE
-			break
 		if(length(GLOB.jobspawn_overrides[rank]))
 			S = pick(GLOB.jobspawn_overrides[rank])
+		else
+			for(var/obj/effect/landmark/start/sloc in GLOB.start_landmarks_list)
```
Comment (spookydonut) [inline code/controllers/subsystem/job.dm]: you can make this loop typeless.

Comment (JJRcop) [reply code/controllers/subsystem/job.dm]: yep!

File: code/modules/mob/living/living.dm [hunk 3]
```diff
@@ -1383,3 +1383,13 @@
...
+		//If you're here from seeing this error, I'm sorry. I'm so very sorry. The error landmark should be a sacred object that nobody has any business messing with, and someone did!
+		//Consider seeing a therapist.
+		CRASH("ERROR-ERROR: ERROR landmark missing!")
```
Comment (ShizCalev) [inline code/modules/mob/living/living.dm]: Log_mapping as well please

**Review by ShizCalev — DISMISSED** [review]

**Review by SpaceManiac — APPROVED** [review]

**Pull request merged**